packages/frontend-2/components/settings/server/user/InviteDialog.vue

```diff
@@ -1,119 +1,161 @@
 import { useInviteUserToServer } from '~~/lib/server/composables/invites'
+import { PlusIcon, TrashIcon } from '@heroicons/vue/24/outline'
+import { isRequired, isEmail } from '~~/lib/common/helpers/validation'
+import { useForm, useFieldArray } from 'vee-validate'
+
+type InviteItem = {
+  email: string
+  serverRole: MaybeNullOrUndefined<ServerRoles>
+  projectId: MaybeNullOrUndefined<string>
+}
 
-const selectedProject = ref(undefined as Optional<FormSelectProjects_ProjectFragment>)
-const serverRole = ref<ServerRoles>(Roles.Server.User)
+enum InputFields {
+  Email = 'email',
+  Role = 'role',
+  Project = 'project'
+}
 
-const { handleSubmit } = useForm<{ message?: string; emailsString: string }>()
+const { handleSubmit } = useForm({
+  initialValues: {
+    invites: [
+      {
+        email: '',
+        serverRole: null,
+        projectId: null
+      }
+    ]
+  }
+})
+const { remove: removeInvite, push: addInvite, fields } = useFieldArray('invites')
 const { mutate: inviteUserToServer } = useInviteUserToServer()
 const inviteUserToProject = useInviteUserToProject()
 const anyMutationsLoading = useMutationLoading()
 const { isAdmin } = useActiveUser()
 const { isGuestMode } = useServerInfo()
 
+const inviteItems = ref<InviteItem[]>([
+  {
+    email: '',
+    serverRole: null,
+    projectId: null
+  }
+])
+
 const isOpen = defineModel<boolean>('open', { required: true })
 
 const allowServerRoleSelect = computed(() => isAdmin.value || isGuestMode.value)
 
 const mp = useMixpanel()
-const onSubmit = handleSubmit(async (values) => {
-  const emails = values.emailsString.split(',').map((i) => i.trim())
-  const project = selectedProject.value
-
-  const success = project
-    ? await inviteUserToProject(
-        project.id,
-        emails.map((email) => ({
-          email,
-          serverRole: allowServerRoleSelect.value ? serverRole.value : undefined
-        }))
-      )
-    : await inviteUserToServer(
-        emails.map((email) => ({
-          email,
-          message: values.message,
-          serverRole: allowServerRoleSelect.value ? serverRole.value : undefined
-        }))
-      )
-  if (success) {
-    isOpen.value = false
-    selectedProject.value = undefined
-    mp.track('Invite Action', {
-      type: 'server invite',
-      name: 'send',
-      multiple: emails.length !== 1,
-      count: emails.length,
-      hasProject: !!project,
-      to: 'email'
-    })
-  }
+
+const onSubmit = handleSubmit(() => {
+  inviteItems.value.forEach(async (invite: InviteItem) => {
+    invite.projectId
+      ? await inviteUserToProject(invite.projectId, [
+          {
+            email: invite.email,
+            serverRole: invite.serverRole
+          }
+        ])
+      : await inviteUserToServer([
+          {
+            email: invite.email,
+            serverRole: invite.serverRole
+          }
+        ])
+  })
+
+  mp.track('Invite Action', {
```

Comment:
@benjaminvo With the new invite modal we have to modify this event, but I'm not sure what makes most sense. Ideally we still want to track wether people are sending 1 or multiple invites out, so we keep `count`, but `hasProject` is now invite specific, but still useful to track. Any suggestion on how to modify/split up this event?

Reply:
Maybe it's best to do one event per invite? It also makes it easier to generate a graph of how many invites are sent.

So `count` is out. It's of course a little interesting to know if multiple invites were sent in the same "session" but I can live without this info.

Reply:
Cool 👍

Reply:
It's indeed intended according to the design :)